### DISCUSSION TITLE
onLoadMore prop on Table.Body component disables auto width feature

### DISCUSSION THREAD

**Discussion by mikehesse** [heroui-inc/heroui#589] [Help]
In my table, I use the [Infinity Pagination]( https://nextui.org/docs/components/table#infinty-pagination) of NextUI. But when I use the table property "onLoadMore" I lose the auto width feature of my columns and the ability to change the width of individual columns. Is this attended or a bug?

I need the auto width or the custom width feature on the columns.
Is there a hidden configuration or a workaround for this?

**Comment by cyanxiao** [comment]
Hi @mikehesse, may I ask what's your expectation about auto width? For example should table adjust columns width once it gets new data when scrolling down?

**Reply by mikehesse** [reply]
Hey @imchell,

Thank you for your message!
I took some screenshots s for a better explanation of my problem:

### Case 1: When I set "onLoadMore" on "Table.Body" and the width of "Table.Column" to a fixed value
<img width="1189" alt="Bildschirmfoto 2022-07-03 um 20 22 24" src="https://user-images.githubusercontent.com/91433744/177052537-1015d6fa-6f68-4c75-9fd3-f8963e1c8f4f.png">
But with this solution, I got much unused white space on the table.

### Case 2: When I set "onLoadMore" and DON'T set the width of "Table.Column"
<img width="1185" alt="Bildschirmfoto 2022-07-03 um 20 22 05" src="https://user-images.githubusercontent.com/91433744/177052538-5c81c909-4b94-4214-90b1-ed728ee8ef30.png">
Here I expect the cell to get measured, like in the last screenshot, but my data is cropped. 


### Case 3: When I set "onLoadMore" and the width of "Table.Column" to a value different between the columns
<img width="1194" alt="Bildschirmfoto 2022-07-03 um 20 22 43" src="https://user-images.githubusercontent.com/91433744/177052536-2baead3c-0a50-48ed-b392-67a46287c50a.png">
As you see in my screenshot, the header has applied the correct width, but the "Table.Body" doesn't. I think this behavior could be a bug.


### Case 4: When I don't set the  "onLoadMore" prop and no width value on "Table.Column"
<img width="1217" alt="Bildschirmfoto 2022-07-03 um 20 23 52" src="https://user-images.githubusercontent.com/91433744/177052533-51c916c6-c7a2-49c6-931a-dc7099169c1b.png">
And this is the result I also expected in case 2.

I hope this helps to understand my problem.

To answer your last question: This would be great, but with a one-time adjustment or the possibility to set widths like in case 3, it would also solve my problem!

**Reply by cyanxiao** [reply]
Thanks for your detailed explanation! I'll look into the issue.

**Reply by mikehesse** [reply]
> Thanks for your detailed explanation! I'll look into the issue.

Hey Chell,
I hope you're doing well so far and I see that nextui is developing fast in a very promising direction!
Do u have any new information on this topic so far or something like a workaround for this behavior?
